src/go/configmanager/config_manager.go

```diff
@@ -138,7 +137,21 @@ func NewConfigManager(mf *metadata.MetadataFetcher, opts options.ConfigGenerator
 
 	}
 
-	if m.serviceConfigFetcher, err = sc.NewServiceConfigFetcher(mf, opts, m.serviceName); err != nil {
+	accessToken := func() (string, time.Duration, error) {
+		// when --non_gcp  is set, instance metadata server(imds) is not defined so
+		// accessToken is unavailable from imds and --service_account_key must be
+		// set to generate accessToken.
+		if mf == nil && opts.ServiceAccountKey == "" {
```

Comment:
We should move this check out of the function.  This check should be done earlier before it is called.